mark.c

```diff
@@ -861,10 +869,13 @@ GC_INNER mse * GC_mark_from(mse *mark_stack_top, mse *mark_stack,
           GC_ASSERT((word)limit >= (word)current_p);
           if (cheri_getaddress(limit) < cheri_getbase(limit)) goto next_object;
 
-          if (cheri_gettag(limit) == 0) {
+          perm = (word)cheri_getperm(limit) & (CHERI_PERM_LOAD
+                                               | CHERI_PERM_STORE
+                                               | CHERI_PERM_EXECUTE);
```

Comment:
`cheri_getperm` returns a `long`, so are we just asking for trouble by casting it to `word`? I don't think there are likely to be so many permissions that it would matter, but it sets off alarm bells.

One option is to make it `int` (in the absence of `bool`):

```c++
    int has_any_rwx;  // No bool in C!
    ...
    has_any_rwx = (cheri_getperm(limit) & (...)) != 0;
```

Reply:
`word` is typecast to `long` in [include/gc.h](https://github.com/djichthys/bdwgc/blob/master/include/gc.h#L70), so I think this should be OK.

Reply:
Also variable name changed in [5d14f76](https://github.com/capablevms/bdwgc/pull/13/commits/5d14f765f99e4339d10dec75739586370510607b).

Reply:
Removed cast in [641ef64](https://github.com/capablevms/bdwgc/pull/13/commits/641ef645fce219a25eeacdbb3a9abcef0b2ae8cd).